lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -3293,6 +3293,11 @@ void __cdecl _alloca_probe_16()
 
         Assert(!(callInfo.Flags & CallFlags_New));
 
+        if (!JavascriptOperators::IsObject(args[0]))
```

Comment:
nit: I think the behavior with this fix is correct, but "should return false ..." wasn't the issue. Original code already returning false for your tests through "!IsCallable" path. Moreover the spec only requires "IsCallable" test (https://tc39.github.io/ecma262/#sec-function.prototype-@@hasinstance, https://tc39.github.io/ecma262/#sec-ordinaryhasinstance), adding "IsObject" test feels a bit unnatural. The issue could be fixed by moving "FromVar" call after "IsCallable".